stdlib/Random/src/Xoshiro.jl

```diff
@@ -117,65 +117,30 @@ end
 
 # Shared implementation between Xoshiro and TaskLocalRNG -- seeding
 
-function seed!(x::Union{TaskLocalRNG,Xoshiro})
+function seed!(rng::Union{TaskLocalRNG,Xoshiro})
     # as we get good randomness from RandomDevice, we can skip hashing
     parent = RandomDevice()
     # Constants have nothing up their sleeve, see task.c
     # 0x02011ce34bce797f == hash(UInt(1))|0x01
     # 0x5a94851fb48a6e05 == hash(UInt(2))|0x01
     # 0x3688cf5d48899fa7 == hash(UInt(3))|0x01
     # 0x867b4bb4c42e5661 == hash(UInt(4))|0x01
-    setstate!(x,
+    setstate!(rng,
               0x02011ce34bce797f * rand(parent, UInt64),
               0x5a94851fb48a6e05 * rand(parent, UInt64),
               0x3688cf5d48899fa7 * rand(parent, UInt64),
               0x867b4bb4c42e5661 * rand(parent, UInt64))
 end
 
-function seed!(rng::Union{TaskLocalRNG,Xoshiro}, seed::NTuple{4,UInt64})
-    # TODO: Consider a less ad-hoc construction
-    # We can afford burning a handful of cycles here, and we don't want any
-    # surprises with respect to bad seeds / bad interactions.
-
-    s0 = s  = Base.hash_64_64(seed[1])
-    s1 = s += Base.hash_64_64(seed[2])
-    s2 = s += Base.hash_64_64(seed[3])
-    s3 = s += Base.hash_64_64(seed[4])
-
+function seed!(rng::Union{TaskLocalRNG,Xoshiro}, seed::Vector{UInt32})
+    c = SHA.SHA2_256_CTX()
+    SHA.update!(c, reinterpret(UInt8, seed))
```

Comment:
This is endian-dependent, which I don't think we want.   Maybe do `hton.(seed)`?

Reply:
Good point, thanks. I will update if this PR gets approved.

Reply:
I don't think we've ever run on a big endian system, and probably never will.